test-tool/test_write_residuals.h

```diff
@@ -0,0 +1,48 @@
+/*
+   Copyright (C) 2013 by Ronnie Sahlberg <ronniesahlberg@gmail.com>
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef _ISCSI_TESTS_WRITE_RESIDUALS_H_
+#define _ISCSI_TESTS_WRITE_RESIDUALS_H_
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+
+#include <CUnit/CUnit.h>
+
+#include "iscsi.h"
+#include "iscsi-private.h"
+#include "scsi-lowlevel.h"
+#include "iscsi-test-cu.h"
+
+
+
+struct residuals_test_data {
+  size_t cdb_size;
+  unsigned int xfer_len;
+  unsigned int buf_len;
+  unsigned int residuals_kind;
+  size_t residuals_amount;
+  bool check_overwrite;
+  const char log_messages[100];
+};
+
+extern bool command_is_implemented;
```

Comment:
What is the purpose of this global variable? Since it is set but not used, can it be removed?

Reply:
This variable is needed because if the command is not implemented, testing should end. It is used in test_write10_residuals.c, test_write12_residuals.c and test_write12_residuals.c
```suggestion
                if (!command_is_implemented) {
                        CU_PASS("WRITE10 is not implemented.");
                        return;
                }
```